kdl_parser/src/kdl_parser.cpp

```diff
@@ -86,8 +93,11 @@ KDL::Joint toKdl(urdf::JointSharedPtr jnt)
         return KDL::Joint(jnt->name, F_parent_jnt.p, F_parent_jnt.M * axis, KDL::Joint::TransAxis);
       }
     default: {
-        ROS_WARN("Converting unknown joint type of joint '%s' into a fixed joint",
-          jnt->name.c_str());
+#ifdef HAS_ROS
+        ROS_WARN("Converting unknown joint type of joint '%s' into a fixed joint", jnt->name.c_str());
+#else
+        std::cerr << "Converting unknown joint type of joint '" << jnt->name << "' into a fixed joint" << std::endl;
```

Comment:
Recommend defining at the top of the file `ROS_WARN`/`ROS_ERROR`/etc instead to keep the number of `HAS_ROS` checks to a minimum.